### PR TITLE
feat: whitelist gifs (no url extension needed)

### LINKED ISSUE #1377: image preview don't work when URL doesn't end in an image extension
e.g `https://static.wikia.nocookie.net/jjba/images/8/83/Beetle_arrow_anime.png/revision/latest/scale-to-width-down/340?cb=20190614221523`

to get around this, we could instead check the url itself contains that extension

### LINKED ISSUE #1479: Add gif site whitelist for image unfurling

### PR DESCRIPTION
**Pull request by emizzle**
Fixes #1377.
Fixes #1479.

Two sites have been added to the whitelist: giphy.com and tenor.com.

`imageUrls` in its entirety has been removed and instead all links are being handle through the message `linkUrls`. This prevents double-handling of urls that may or may not be images.

The logic to automatically show links previews works like this:
1. If the setting "display chat images" is enabled, all links that *contain* ".png", ".jpg", ".jpeg", ".svg", ".gif" will be automatically shown. If the URL doesn't contain the extension, we are not downloading it, unless it is whitelisted. This was meant to be somewhat of a security compromise as we do not want to download each and every link posted in a message just to find out its true content type. Obviously, it's still unsafe to download any link containing an image extension, but that's why this is a user-enabled option.
2. If the above setting is *disabled*, then we follow the whitelist settings for tenor and giphy (and youtube). This allows us to preview gifs that do not have a file extension in their url.

feat: bump status-go to the commit that supports the new whitelist (https://github.com/status-im/status-go/pull/2094), and also lets us get link preview data from urls in the whitelist. NOTE: this commit was branched off status-go `develop`, so once it is merged, and we update this PR to the new commit, we will effectively be getting status-go develop changes. We *could* base that status-go PR off of master if it makes things easier.

fix: height on settings update issue

feat: move date/time of message below links

fix: layout issues when changing setting `neverAskAboutUnfurlingAgain`

feat: Add MessageBorder component to aid in showing rounded corners with different radius

## Notes
The `LinksMessage` component is loaded for each and every message containing `linkUrls`. Then if those links are images (depending on user settings and whitelist - see above for more info), we download the image. When scrolling through lots of messages, there is a very noticeable "stutter", and I'm concerned that this is due to the images being downloaded and possibly the logic of `LinksMessage` (which access `appSettings`). We may need to find a way to offload the downloading of images to a new thread, or only download images once scrolling has stopped for a period (like 100ms).

### CODE REVIEW

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageLoader.qml [hunk 1]
```diff
@@ -9,7 +9,7 @@ Item {
     property bool isCurrentUser: false
     property url source
     property bool playing: true
-    property bool isAnimated: !!source && source.toString().endsWith('.gif')
+    property bool isAnimated: true
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageLoader.qml:12]: Is this a leftover of a test or we start it as true and change it later?

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml [hunk 2]
```diff
@@ -1,59 +0,0 @@
-import QtQuick 2.3
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:1]: RIP ImageMessage

File: ui/app/AppLayouts/Profile/Sections/PrivacyContainer.qml [hunk 3]
```diff
@@ -153,23 +132,35 @@ Item {
...
-
         RowLayout {
-            id: displayImageSettings
-            StyledText {
-                //% "Display images in chat automatically"
-                text: qsTrId("display-images-in-chat-automatically")
+            spacing: Style.current.padding
+            width: parent.width
+            Column {
+                // width: parent.width - (displayChatImagesSwitch.width + Style.current.padding)
```
Comment (jrainville) [inline ui/app/AppLayouts/Profile/Sections/PrivacyContainer.qml:139]:
```suggestion
```

File: ui/main.qml [hunk 4]
```diff
@@ -162,20 +162,43 @@ ApplicationWindow {
...
+                        settingsUpdated = true
+                    }
+                })
+                if (settingsUpdated) {
+                    appSettings.whitelistedUnfurlingSites = settings
+                }
+            } catch (e) {
+                console.error('Could not parse the whitelist for sites', e)
+            }
+            console.log(">>> [Main] firing settings loaded")
```
Comment (jrainville) [inline ui/main.qml:200]: Is this log needed?

**Review by jrainville — COMMENTED** [review]
Nice work!
Just had so cleanup comments

File: ui/app/AppLayouts/Profile/Sections/PrivacyContainer.qml (hunk 3 above)
Comment (emizzle) [reply ui/app/AppLayouts/Profile/Sections/PrivacyContainer.qml:139]: Fixed.

File: ui/main.qml (hunk 4 above)
Comment (emizzle) [reply ui/main.qml]: Fixed.

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageLoader.qml (hunk 1 above)
Comment (emizzle) [reply ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageLoader.qml:12]: It is changed later (when image is clicked), but it default to false for all images. According to the documentation, this is the default anyway, and the `AnimatedImage` control has the ability to handle static images on its own anyway. 

https://doc.qt.io/qt-5/qml-qtquick-animatedimage.html#playing-prop

Additionally, this allows us to support animated image types that aren't necessarily limited to images that end in a `.gif` extension.

**Review by jrainville — APPROVED** [review]

**Pull request merged**